src/app/features/postsSlice.jsx

```diff
@@ -3,62 +3,118 @@ import { calcTime } from "../helpers/helps";
-                numberOfComments:num_comments,
-                id:id
+        try {
+            const response = await fetch(`https://www.reddit.com/r/${subreddit}/${filter}.json`);
+            if (!response.ok) {
+                throw new Error('Server Error!');
             }
-        });
-        return newPosts;
-        
+            const jsonData = await response.json();
+            const newPosts = jsonData.data.children.map((post) => {
+                const { subreddit_name_prefixed, author, id, num_comments, title } = post.data;
+                const postText = post.data.selftext ? post.data.selftext : null;
+                let postImage = post.data.url;
+                if (postImage.includes('.jpg') || postImage.includes('.png')) {
+                } else {
+                    postImage = null;
+                };
+                const timestamp = post.data.created_utc;
+                const time = calcTime(timestamp);
+                return {
+                    author: author,
+                    subreddit: subreddit_name_prefixed,
+                    time: time,
+                    title: title,
+                    text: postText,
+                    image: postImage,
+                    numberOfComments: num_comments,
+                    id: id
+                }
+            });
+            return newPosts;
+        } catch (error) {
+            return rejectWithValue(error.message);
+        }
+
+
+    }
+);
+
+export const fetchPostsBySearchPosts = createAsyncThunk(
+    'posts/fetchPostsBySearchPosts',
+    async (searchTerm, { rejectWithValue }) => {
+        try {
+            const response = await fetch(`https://www.reddit.com/search.json?q=${searchTerm}`);
+            if (!response.ok) {
+                throw new Error('Not found!');
+            }
+            const jsonData = await response.json();
+            const newPosts = jsonData.data.children.map((post) => {
+                const { subreddit_name_prefixed, author, id, num_comments, title } = post.data;
+                const postText = post.data.selftext ? post.data.selftext : null;
+                let postImage = post.data.url;
+                if (postImage.includes('.jpg') || postImage.includes('.png')) {
+                } else {
+                    postImage = null;
+                };
+                const timestamp = post.data.created_utc;
+                const time = calcTime(timestamp);
+                return {
+                    author: author,
+                    subreddit: subreddit_name_prefixed,
+                    time: time,
+                    title: title,
+                    text: postText,
+                    image: postImage,
+                    numberOfComments: num_comments,
+                    id: id
+                }
+            });
+            return newPosts;
+        } catch (error) {
+            return rejectWithValue(error.message);
+        }
+
+
     }
+
```

Comment:
Queste due funzioni sono praticamente identiche. Devi sempre tenere in mente i concetti DRY (don;t repeat yourself).
Devi valutare quali sono le parti del codice che puoi estrarre in funzioni riutilizzabili. 

- Nella chiamata che avevamo avuto mi ricordo d'averti consigliato di estrarre il la parte del map in una funzione separata.